app/assets/javascripts/components/ops/tenant-component.js

```diff
@@ -0,0 +1,94 @@
+ManageIQ.angular.app.component('tenantComponent', {
+  bindings: {
+    recordId: '=?',
+    divisible: '=',
+    parent: '=?',
+    redirectUrl: '@',
+  },
+  controllerAs: 'vm',
+  controller: tenantFormController,
+  templateUrl: '/static/ops/tenant/tenant.html.haml',
+});
+
+tenantFormController.$inject = ['API', 'miqService'];
+
+function tenantFormController(API, miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.entity = vm.divisible ? __('Tenant') : __('Project');
+    vm.saveable = miqService.saveable;
+    vm.afterGet = false;
+
+    vm.tenantModel = {
+      name: '',
+      description: '',
+      parent: vm.parent,
+      use_config_for_attributes: '',
+      default: '',
+    };
+
+    if (angular.isDefined(vm.recordId)) {
+      vm.newRecord = false;
+      miqService.sparkleOn();
+      API.get('/api/tenants/' + vm.recordId)
+        .then(getTenantFormData)
+        .catch(miqService.handleFailure);
+    } else {
+      vm.newRecord = true;
+      vm.afterGet = true;
+      vm.modelCopy = angular.copy( vm.tenantModel );
+    }
+
+    vm.saveClicked = function() {
```

Comment:
Just a detail, but these functions should be defined outside `$onInit`, that's just for code to run on init..